src/ai/whylabs/templates/batch_bigquery_template.py

```diff
@@ -30,30 +43,31 @@ class TemplateArgs:
     logging_level: str
     date_column: str
     date_grouping_frequency: str
+    segment_columns: Optional[str]
 
 
-class ViewCombiner(beam.CombineFn):
-    def __init__(self, args: TemplateArgs):
-        self.logger = logging.getLogger("ViewCombiner")
-        self.logging_level = args.logging_level
+@dataclass
+class InputBigQuerySQL:
+    query: str
 
-    def setup(self) -> None:
-        self.logger.setLevel(logging.getLevelName(self.logging_level))
 
-    def create_accumulator(self) -> DatasetProfileView:
-        return DatasetProfile().view()
+@dataclass
+class InputBigQueryTable:
+    table_spec: str
 
-    def add_input(self, accumulator: DatasetProfileView, input: DatasetProfileView) -> DatasetProfileView:
-        return accumulator.merge(input)
 
-    def merge_accumulators(self, accumulators: List[DatasetProfileView]) -> DatasetProfileView:
-        view: DatasetProfileView = DatasetProfile().view()
-        for current_view in accumulators:
-            view = view.merge(current_view)
-        return view
+@dataclass
+class InputOffset:
+    offset: int
+    table_spec: str
+    timezone: tzinfo
+    query: str
 
-    def extract_output(self, accumulator: DatasetProfileView) -> DatasetProfileView:
-        return accumulator
+
+class SegmentDefinition(NamedTuple):
+    segment_column: str
+    segment: Segment
```

Comment:
@jamie256 we're confused about what the tuple represents in this object. It seems to just be a single value with the way that we're using segmentation here.